### PR TITLE
fix(azure_blob sink): Base Content-Type on encoder and not compression

### PR DESCRIPTION
**Pull request by stemjacobs**
## Description
Update setting of the content type associated with a blob to be the uncompressed underlying encoded type.

## Issues

The following open issues are concerned with this PR: 
Closes: #18183

### CODE REVIEW

File: src/sinks/azure_blob/integration_tests.rs [hunk 1]
```diff
@@ -170,9 +181,13 @@ async fn azure_blob_insert_json_into_blob_gzip() {
...
     assert!(blobs[0].clone().ends_with(".log.gz"));
     let (blob, blob_lines) = config.get_blob(blobs[0].clone()).await;
+    assert_eq!(
+        blob.properties.content_encoding,
+        Some(String::from("gzip"))
+    );
     assert_eq!(
         blob.properties.content_type,
-        String::from("application/gzip")
+        String::from("application/json")
```
Comment (dsmith3197) [inline src/sinks/azure_blob/integration_tests.rs:190]: I *think* the content-type here will still be `application/x-ndjson` due to the newline delimited framer in the encoder.
```suggestion
        String::from("application/x-ndjson")
```

**Review by dsmith3197 — CHANGES_REQUESTED** [review]
Looks good, will approve once comment is addressed.

File: src/sinks/azure_blob/integration_tests.rs (hunk 1 above)
Comment (stemjacobs) [reply src/sinks/azure_blob/integration_tests.rs]: Good callout, overlooked this and the related test was ignored.

**Review by dsmith3197 — DISMISSED** [review]

**Comment by dsmith3197** [comment]
@stemjacobs, could you format the code by running `cargo vdev fmt`? Thanks 🙂 

That's the reason checks are failing
```
cargo vdev check fmt
File "src/sinks/azure_blob/integration_tests.rs" contains trailing spaces in some of the lines
File "src/sinks/azure_blob/request_builder.rs" doesn't end with a newline
Error: command: cd "/home/runner/work/vector/vector" && "/home/runner/work/vector/vector/scripts/check-style.sh"
  failed with exit code: 1
make: *** [Makefile:453: check-fmt] Error 1
```

**Review by dsmith3197 — APPROVED** [review]

**Pull request merged**